### ISSUE TITLE
Aria-sort is not working for tables with pagination enabled

### ISSUE THREAD

**Issue by luizas** [nvaccess/nvda#17732]
When the aria-sort attribute is applied to the table header to indicate the sorting order, the screen reader fails to announce 'sorting ascending' or 'sorting descending' as expected. This issue specifically arises in cases where the table includes pagination. Despite the pagination, the entire dataset is sorted, not just the rows that are currently visible on the screen. The lack of proper announcements by the screen reader for the sorting state (ascending or descending) makes it difficult for users relying on assistive technologies to understand the current sorting order of the entire table, especially when navigating across multiple pages of content

### Steps to reproduce:

1. Open https://codepen.io/luizas/pen/gbYmWdY   (Edge).
3. Run NVDA.
4. Press on header (First Name) using the mouse, spacebar, or Enter key.
5. Listen to the screen reader.

### Actual behavior:
The screen reader reads: 'First Name.'

### Expected behavior:
Screen reader reads - Sorting ascending / Sorting descending.

#### NVDA version:
2024.4.1

#### Windows version: 
Windows 11 Enterprise

#### Other information about your system:
Edge:Version 133.0.3065.82 (Official build) (64-bit)

#### Does the issue still occur after restarting your computer?
yes

**Comment by gerald-hartig** [comment]
Closing issue. This is an HTML authoring issue. ARIA sort is being set on the button inside the TH and you're changing it on the button. It should only be set on the row or column header.

**Comment by luizas** [comment]
@gerald-hartig  ARIA sort is being set on TH not on the button.  [](https://codepen.io/luizas/pen/gbYmWdY)  https://codepen.io/luizas/pen/gbYmWdY

     <th aria-sort="ascending"><button>First Name</button></th>
          <th aria-sort="ascending"><button>Last Name</button></th>

**Comment by SaschaCowley** [comment]
@luizas it is being set *once* on the `th`, but changed by JS on the `button`. Copied directly from Firefox dev tools:
```html
<th aria-sort="ascending"><button aria-sort="descending">First Name</button></th>
```

**Comment by luizas** [comment]
Hi @SaschaCowley, could you please take another look at this: https://codepen.io/luizas/pen/gbYmWdY?
We’re seeing a similar issue in a large application — sorting the entire table works with JAWS, but not with NVDA.
